### PR TITLE
Update meshmodel search by name

### PR DESCRIPTION
**Pull request by abdullah1308**
**Notes for Reviewers**

This PR fixes #

Updated Meshmodel model and component search to be consistent across all Meshmodel APIs

Searching a model or a component is done through its `displayName` field.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by abdullah1308** [comment]
@UsmanDev09 FYI

File: server/handlers/component_handler.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ func (h *Handler) GetMeshmodelModelsByCategories(rw http.ResponseWriter, r *http
 	}
 	if r.URL.Query().Get("search") != "" {
 		filter.Greedy = true
-		filter.Name = r.URL.Query().Get("search")
+		filter.DisplayName = r.URL.Query().Get("search")
```
Comment (leecalcote) [inline server/handlers/component_handler.go:65]: @Abhishek-kumar09 this works well with MeshMap, or no?

Comment (leecalcote) [reply server/handlers/component_handler.go:65]: @abdullah1308 maybe you can confirm.

Comment (abdullah1308) [reply server/handlers/component_handler.go:65]: @leecalcote I've checked out the meshmap code, these changes should work well with Meshmap.

Comment (abdullah1308) [reply server/handlers/component_handler.go:65]: The endpoints changed in this PR aren't being used anywhere. The changes made are to make the search functionality in these endpoints inline with the endpoints that are being used.

**Pull request merged**